src/mlpack/methods/ann/loss_functions/multilabel_softmargin_loss_impl.hpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file multilabel_softmargin_loss_impl.hpp
+ * @author Anjishnu Mukherjee
+ *
+ * Implementation of the Multi Label Soft Margin Loss function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_MULTILABEL_SOFTMARGIN_LOSS_IMPL_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_MULTILABEL_SOFTMARGIN_LOSS_IMPL_HPP
+
+// In case it hasn't been included.
+#include "multilabel_softmargin_loss.hpp"
+
+namespace mlpack {
+namespace ann /** Artifical Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+MultiLabelSoftMarginLoss<InputDataType, OutputDataType>::
+MultiLabelSoftMarginLoss(
+    const bool reduction,
+    const arma::rowvec& weights) :
+    reduction(reduction),
+    weighted(false)
+{
+  if (weights.n_elem)
+  {
+    classWeights = weights;
+    weighted = true;
+  }
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType>
+typename InputType::elem_type
+MultiLabelSoftMarginLoss<InputDataType, OutputDataType>::Forward(
+    const InputType& input, const TargetType& target)
+{
+  if (!weighted)
+    classWeights.ones(1, input.n_cols);
+
+  InputType logSigmoid = arma::log((1 / (1 + arma::exp(-input))));
+  InputType logSigmoidNeg = arma::log(1 / (1 + arma::exp(input)));
```

Comment:
Hey @iamshnoo, it seems like this can cause `nan` problem for large values. You can refer to #2371. Amphan has turned pretty dangerous 😢. Hope you are safe Anjishnu.

Reply:
Regarding the point you raised about the calculation of ```logSigmoid``` and ```logSigmoidNeg```, quite valid points indeed. If I was expecting input to be a large enough number, this raises the possibility of giving ```nan```. But, this being a loss function and not an activation function(unlike in PR 2371) is applied at the end of a forward pass and almost always in conjunction with a softmax layer, which will keep all the values in input in the range of 0 to 1. So, I don't think we need to bother at all about ```input``` being a large number for this case. 

Let me know what you think about this! I will get back as and when I can manage some network.

Reply:
That makes sense. Thanks for the clarification.